kata-deploy/action/test-kata.sh

```diff
@@ -0,0 +1,163 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+function die() {
+	msg="$*"
+	echo "ERROR: $msg" >&2
+	exit 1
+}
+
+function waitForProcess() {
+        wait_time="$1"
+        sleep_time="$2"
+        cmd="$3"
+        while [ "$wait_time" -gt 0 ]; do
+                if eval "$cmd"; then
+                        return 0
+                else
+						echo "waiting"
+                        sleep "$sleep_time"
+                        wait_time=$((wait_time-sleep_time))
+                fi
+        done
+        return 1
+}
+
+# waitForLabelRemoval will wait for the kata-runtime labels to removed until a given
+# timeout expires
+function waitForLabelRemoval() {
+  wait_time="$1"
+  sleep_time="$2"
+
+  while [[ "$wait_time" -gt 0 ]]; do
+    # if a node is found which matches node-select, the output will include a column for node name,
+    # NAME. Let's look for that 
+    if [[ -z $(kubectl get nodes --selector katacontainers.io/kata-runtime | grep NAME) ]]
+    then
+        return 0
+    else
+        echo "waiting for kata-runtime label to be removed"
+        sleep "$sleep_time"
+        wait_time=$((wait_time-sleep_time))
+    fi
+  done
+
+  echo "failed to cleanup"
+  return 1
+}
+
+
+function run_test() {
+	PKG_SHA=$1
+	YAMLPATH="https://raw.githubusercontent.com/amshinde/kata-packaging/$PKG_SHA/kata-deploy"
+	echo "verify connectivity with a pod using Kata"
+
+	deployment=""
+	busybox_pod="test-nginx"
+	busybox_image="busybox"
+	cmd="kubectl get pods | grep $busybox_pod | grep Completed"
+	wait_time=120
+	sleep_time=3
+
+    configurations=("nginx-deployment-qemu" "nginx-deployment-qemu-virtiofs")
+    for deployment in "${configurations[@]}"; do
+	  # start the kata pod:
+	  kubectl apply -f "$YAMLPATH/examples/${deployment}.yaml"
+
+	  # in case the control plane is slow, give it a few seconds to accept the yaml, otherwise
+	  # our 'wait' for deployment status will fail to find the deployment at all
+	  sleep 3 
```

Comment:
Why?

Reply:
because azure takes a long time to persist it into etcd.